coderadar-vcs/src/main/java/io/reflectoring/coderadar/vcs/adapter/ExtractProjectCommitsAdapter.java

```diff
@@ -49,26 +46,31 @@
   }
 
   private List<Commit> getCommits(Git git, DateRange range) throws GitAPIException {
-    Map<ObjectId, Commit> map = new HashMap<>();
-    List<Commit> result = new ArrayList<>();
     List<RevCommit> revCommits = new ArrayList<>();
     git.log().call().iterator().forEachRemaining(revCommits::add);
-
-    for (int i = revCommits.size() - 1; i >= 0; --i) {
+    int revCommitsSize = revCommits.size();
+    List<Commit> result = new ArrayList<>(revCommitsSize);
+    IdentityHashMap<RevCommit, Commit> map = new IdentityHashMap<>(revCommitsSize);
+    for (int i = revCommitsSize - 1; i >= 0; --i) {
       RevCommit rc = revCommits.get(i);
       if (isInDateRange(range, rc)) {
-        Commit commit = map.computeIfAbsent(rc.getId(), objectId -> mapRevCommitToCommit(rc));
+        Commit commit = map.computeIfAbsent(rc, objectId -> mapRevCommitToCommit(rc));
+        List<Commit> parents =
+            rc.getParentCount() > 0
```

Comment:
Does the `getParentCount` method only return 0 for the first Commit in a repository?

Reply:
I don't know. Technically, not always.